File: 122.best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.java
public class Solution {
    public int maxProfit(int[] prices) {
    	if (prices.length == 0)
			return 0;
		int res = 0;
		int max = prices[0];
		for (int i = 0; i < prices.length; ++i) {
			if (prices[i] > max) {
				res += prices[i] - max;
			}
			max = prices[i];
		}
		return res;
    }
}

File: 122.best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
    	if(prices.size() == 0) return 0;
    	int res = 0;
    	int max = prices.front();
        for(vector<int>::iterator it = prices.begin(); it != prices.end(); ++it) {
        	if(*it > max) {
        		res += *it - max;
        	}
        	max = *it;
        }
        return res;
    }
};